docs/JSONDocumentation.md

```diff
@@ -0,0 +1,379 @@
+| Properties            | array of [Property](#property)                   |                                                      | Not applicable                                      |
+| Attributes            | array of [Attribute](#attribute)                 |                                                      | Annotations                                         |
+| EnumMembers           | array of [EnumMember](#enummember)               | Enumeration constant                                 | Enumeration constant                                |
+| Events                | array of [Event](#event)                         |                                                      | Not applicable                                      |
+| DocumentationComments | [DocumentationComments](#documentation-comments) | XML documentation comments                           | JavaDoc                                             |
+
+## Modifiers
+Different modifiers (access, non-access, other) are combined into a single value using a bitmap. Currently, modifiers that exist only in Java but not in .NET are not included. These could be added in the future.
+
+The following modifiers are included:
+
+| .NET Modifier | Bitmap Value | Java Equivalent                               |
+|---------------|--------------|-----------------------------------------------|
+| Internal      | 1 << 0       | None                                          |
+| Public        | 1 << 1       | Public                                        |
+| Private       | 1 << 2       | Private                                       |
+| Protected     | 1 << 3       | Protected                                     |
+| Static        | 1 << 4       | Static                                        |
+| Abstract      | 1 << 5       | Abstract                                      |
+| Override      | 1 << 6       | N/A                                           |
+| Readonly      | 1 << 7       | Final (when referring to fields or variables) |
+| Async         | 1 << 8       | N/A                                           |
+| Const         | 1 << 9       | N/A                                           |
+| Sealed        | 1 << 10      | Final (when referring to classes)             |
+| Virtual       | 1 << 11      | N/A                                           |
+| Extern        | 1 << 12      | N/A                                           |
+| New           | 1 << 13      | N/A                                           |
+| Unsafe        | 1 << 14      | N/A                                           |
+| Partial       | 1 << 15      | N/A                                           |
+
+
+## Field
+
+### Required keys:
+
+| Key  | Type   | .NET              | Java              |
+|------|--------|-------------------|-------------------|
+| Name | string | Name of the field | Name of the field |
+| Type | string | Type of the field | Type of the field |
```

Comment:
Here and below, when mentioning types, it wouldn't hurt to mention that these strings are supposed to be fully qualified - otherwise, people might expect this to match how it shows up in the source, but it does not (we give `java.lang.String`, not `String`).